### ISSUE TITLE
Extra extension added to filenames with an existing extension when markdown_link_ext is set

### ISSUE THREAD

**Issue by folofjc** [vimwiki/vimwiki#1256]
Prior to submitting a new issue make sure to complete these steps:

- [x ] Checkout the `dev` branch and confirm the issue is present there as well.
      The `dev` branch contains fixes that may not have been merged to `master` yet.
- [x ] Post the syntax you are using (default/mediawiki/markdown) **and** your vimwiki settings from your `.vimrc`
- [x ] Provide a detailed description of the problem including **steps to reproduce the issue**.
- [x ] Include the output of `:VimwikiShowVersion`.

Version:
```
Os: Linux
Vim: 802
Branch: dev
Revision: fc2e9f0
Date: 2022-07-18 20:52:08 -0600 
```

vimrc settings:
```
let g:vimwiki_markdown_link_ext = 1
```

I just moved to `dev` because `master` is [super old](https://github.com/vimwiki/vimwiki/issues/831#issuecomment-1180595732). 

On `master`, when my links all contain the `.md` extension (because of the above `~/.vimrc` setting), pressing enter on a link takes me to the page.

On `dev`, when my links have the `.md` extension and I press enter on a link, it adds another `.md` so tries to open a page like `Testfile.md.md` which does not exist.

**Comment by folofjc** [comment]
> Noting that [4672deb](https://github.com/vimwiki/vimwiki/commit/4672deb44e34b8e779e6ac8c51050140fac78c66) moved that to a wikilocal setting without updating the docs, although I think it inherits the global setting.

Ah, okay, yeah I did not see this in the docs. I went back to `master` for the time being.

**Comment by codevion** [comment]
Likely https://github.com/vimwiki/vimwiki/commit/d7ec12645a0460a7d200279c52915e6e080e9869 broke this. Now I'm stuck in a weird place where 

On master -> 
`[10.02.blah](10.02.blah.md)` -> takes me to -> `10.02.blah` file (no file extension at all)

On dev
`[10.02.blah](10.02.blah.md)` -> takes me to -> `10.02.blah.md.md` file

So I have to use dev and remove all the .md if I want to go to the real file :( 

Btw, this is with `markdown_link_ext = 1,` set as a wiki local config (in lua though)

**Comment by folofjc** [comment]
> Btw, this is with `markdown_link_ext = 1,` set as a wiki local config (in lua though)

Yes, but I think if you change that to a global setting, you can go to master and be fine, right?

**Comment by codevion** [comment]
That doesn't work because master has the issue https://github.com/vimwiki/vimwiki/issues/950 that the fix I linked was trying to solve. Everything would work fine in master if I didn't use .'s in my link names.

**Comment by brennen** [comment]
> Likely d7ec126 broke this.

Seems plausible.  I'll see if I can figure out what's going on.

**Comment by brennen** [comment]
(Also, sorry about that.)

**Comment by brennen** [comment]
Think I got it - testing of #1257 to see if it fixes the issues here would be welcome.

**Comment by jokoso** [comment]
@brennen, I just tested out your fix on`brennen-dot-ext` and it seems to fix the issue with the double `.md` extension. :tada: 

```
Os: Linux
Vim: 800 (using neovim master build c223875 from Tue Aug 2 11:52:33 2022 +0200)
Branch: brennen-dot-ext
Revision: 6272912
Date: 2022-07-22 23:04:45 -0600
```

Thanks for the quick fix!

**Comment by brennen** [comment]
Seems fixed.

**Comment by psvelayudhan** [comment]
Hi, I am not sure if this is an intentional change in functionality but I have noticed that on 7cfc3eee2291cec5286e554c29100486a47ee27f, I was very easily able to link to various different filetypes (e.g. .py, .Rmd, etc.).

When d7ec12645a0460a7d200279c52915e6e080e9869 happened this functionality was lost (as is known in this thread) and links to these files would just become *.py.md or *.Rmd.md (or even *.md.md if the .md was written out in the link).

The fix in c21c939b0630f5f453e85621cf611a5b86537511 appears to have resolved this issue for links that explicitly ended in .md (no more .md.md), but the other filetypes I would link to are still struggling (.Rmd files are all linking to .Rmd.md instead). I can provide a reprex and more details about my configuration if needed but I suspect this will be easily reproduced.

If the ability to link to these other files was not desired behaviour I understand but I must say it was quite convenient.
